### PR TITLE
Generate and validate safe module, helm and docker names in create cmd

### LINKED ISSUE #21: Create command should not allow module names with dashes and special characters in module, helm and docker names
Python module names should be letters, numbers separated by underscores only. 
We should also validate the helm_name and docker_name.

### PR DESCRIPTION
**Pull request by dvaccarosenna**
Default module, helm and docker names are generated with incompatible characters removed from the app name. 

All user supplied module, helm and docker names are validated for incompatible characters. 

Closes #21

### CODE REVIEW

File: ignition/templates/factory.py [hunk 1]
```diff
@@ -32,26 +33,47 @@ def __init__(self, driver_types, app_name, version="1.0.0", port=None, module_na
...
+    
     def generate_helm_name(self, app_name):
-        return app_name.lower().replace(" ", "-")
+        filtered_app_name = re.sub('[^A-Za-z0-9-_ ]+', '', app_name)
+        filtered_app_name = " ".join(filtered_app_name.split())
+        return filtered_app_name.lower().replace(" ", "-")
 
     def generate_docker_name(self, app_name):
-        return app_name.lower().replace(" ", "-")
+        filtered_app_name = re.sub('[^A-Za-z0-9-_ ]+', '', app_name)
```
Comment (sglover) [inline ignition/templates/factory.py]: minor: app_name -> docker_name?

File: ignition/templates/factory.py [hunk 2]
```diff
@@ -32,26 +33,47 @@ def __init__(self, driver_types, app_name, version="1.0.0", port=None, module_na
...
+        filtered_app_name = " ".join(filtered_app_name.split())
+        return filtered_app_name.lower().replace(" ", "-")
 
     def generate_module_name(self, app_name):
-        return app_name.lower().replace(" ", "").replace("-", "").replace("_", "")
+        filtered_app_name = re.sub('[^A-Za-z0-9 ]+', '', app_name)
```
Comment (sglover) [inline ignition/templates/factory.py]: minor: app_name -> module_name?

File: ignition/templates/factory.py [hunk 3]
```diff
@@ -32,26 +33,47 @@ def __init__(self, driver_types, app_name, version="1.0.0", port=None, module_na
...
+        if not re.match('^[a-zA-Z0-9-_]*$', helm_name):
+            raise ValueError('helm_name must be a string with characters from a-z, A-Z, 0-9, dash (-) or underscore (_) but was: {0}'.format(helm_name))
+
+    def __validate_docker_name(self, docker_name):
+        if not re.match('^[a-zA-Z0-9-_]*$', docker_name):
+            raise ValueError('docker_name must be a string with characters from a-z, A-Z, 0-9, dash (-) or underscore (_) but was: {0}'.format(docker_name))
+    
     def generate_helm_name(self, app_name):
-        return app_name.lower().replace(" ", "-")
+        filtered_app_name = re.sub('[^A-Za-z0-9-_ ]+', '', app_name)
```
Comment (sglover) [inline ignition/templates/factory.py]: minor: app_name -> helm_name?

Comment (dvaccarosenna) [reply ignition/templates/factory.py]: Well the app_name is the input, but I'll change the "filtered_app_name" to show it's no longer the app_name but the helm_name

**Pull request merged**